### PR TITLE
CB-18482 Create flow in environment service for proxy edit

### PR DESCRIPTION
**Pull request by Bajzathd**
Currently the only meaningful step is to save the new proxy association to the db, additional steps will be added in subsequent tickets.

See detailed description in the commit message.

### CODE REVIEW

**Review by pkedvessy — APPROVED** [review]

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/modify/proxy/config/EnvProxyModificationFlowConfig.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+public class EnvProxyModificationFlowConfig extends
+        AbstractFlowConfiguration<EnvProxyModificationState, EnvProxyModificationStateSelectors>
+        implements RetryableFlowConfiguration<EnvProxyModificationStateSelectors> {
+
+    private static final List<Transition<EnvProxyModificationState, EnvProxyModificationStateSelectors>> TRANSITIONS =
+            new Transition.Builder<EnvProxyModificationState, EnvProxyModificationStateSelectors>()
+                    .defaultFailureEvent(FAILED_MODIFY_PROXY_EVENT)
+
+                    .from(INIT_STATE).to(PROXY_CONFIG_MODIFICATION_START_STATE)
+                    .event(EnvProxyModificationStateSelectors.MODIFY_PROXY_START_EVENT).defaultFailureEvent()
```
Comment (bergerdenes) [inline environment/src/main/java/com/sequenceiq/environment/environment/flow/modify/proxy/config/EnvProxyModificationFlowConfig.java:30]: Please be conform with other flow configs and do a fully qualified import for these enum values.

File: environment/src/test/java/com/sequenceiq/environment/proxy/v1/service/ProxyConfigServiceTest.java [hunk 2]
```diff
@@ -204,4 +211,71 @@ public void testDeleteMultipleInAccountHasAllResult() {
...
+        Environment environment = new Environment();
+        ProxyConfig newProxyConfig = new ProxyConfig();
+
+        boolean result = underTestProxyConfigService.shouldModify(environment, newProxyConfig);
+
+        Assertions.assertFalse(result);
+    }
+
+    @Test
+    void shouldModifySameProxies() {
```
Comment (bergerdenes) [inline environment/src/test/java/com/sequenceiq/environment/proxy/v1/service/ProxyConfigServiceTest.java:233]: naming can be misleading as it should _not_ modify

File: environment/src/test/java/com/sequenceiq/environment/proxy/v1/service/ProxyConfigServiceTest.java [hunk 3]
```diff
@@ -204,4 +211,71 @@ public void testDeleteMultipleInAccountHasAllResult() {
...
+
+    @Test
+    void validateModify() {
+        EnvironmentDto environmentDto = EnvironmentDto.builder().build();
+        underTestProxyConfigService.validateModify(environmentDto);
+        verify(proxyConfigModificationService).validateModify(environmentDto);
+    }
+
+    @Test
+    void shouldModifyNoProxies() {
```
Comment (bergerdenes) [inline environment/src/test/java/com/sequenceiq/environment/proxy/v1/service/ProxyConfigServiceTest.java:223]: naming can be misleading as it should _not_ modify

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/modify/proxy/handler/EnvProxyModificationSaveAssociationHandler.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public String selector() {
+        return EnvProxyModificationHandlerSelectors.SAVE_NEW_PROXY_ASSOCIATION_HANDLER_EVENT.selector();
+    }
+
+    @Override
+    public void accept(Event<EnvProxyModificationDefaultEvent> event) {
+        EnvProxyModificationDefaultEvent eventData = event.getData();
+        EnvironmentDto environmentDto = environmentService.updateProxyConfig(eventData.getEnvironmentDto().getId(), eventData.getProxyConfig());
```
[the remark this answers is not in the document]
Comment (Bajzathd) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/modify/proxy/handler/EnvProxyModificationSaveAssociationHandler.java:33]: AbstractEnvProxyModificationAction.getFailurePayload should create a failed event in that case, that is handled by ProxyConfigModificationFailedStateAction

[the remark this answers is not in the document]
Comment (bergerdenes) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/modify/proxy/handler/EnvProxyModificationSaveAssociationHandler.java:33]: ok

[the remark this answers is not in the document]
Comment (Bajzathd) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/modify/proxy/handler/EnvProxyModificationSaveAssociationHandler.java]: fixed it with the other comments

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**